File: ToDo/app/src/main/java/com/example/to_do/Adapter/ToDoAdapter.java
package com.example.to_do.Adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;

import androidx.recyclerview.widget.RecyclerView;

import com.example.to_do.MainActivity;
import com.example.to_do.Model.ToDoModel;
import com.example.to_do.R;

import java.util.List;

public class ToDoAdapter extends RecyclerView<ToDoAdapter.ViewHolder> {
    private List<ToDoModel> todoList;
    private MainActivity activity;

    public ToDoAdapter(MainActivity activity){
        this.activity = activity;
    }

    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.task_layout, parent, false);
        return new ViewHolder(itemView);
    }

    public void onBindViewHolder(ViewHolder holder, int position){
        ToDoModel item = todoList.get(position);
        holder.task.setText(item.getTask());
        holder.task.setChecked(toBoolean(item.getStatus()));
    }

    public int getItemCount(){
        return todoList.size();

    }

    private boolean toBoolean(int n){
        return n != 0;
    }

    public void setTasks(List<ToDoModel> todolist){
        this.todoList = todolist;
        notifyDataSetChanged();
    }
    public static class ViewHolder extends RecyclerView.ViewHolder {
        CheckBox task;

        ViewHolder(View view){
            super(view);
            task = view.findViewById(R.id.todoCheckBox);
        }
    }
}

File: ToDo/app/src/main/java/com/example/to_do/Adapter/ToDoAdapter.kt
package com.example.to_do.Adapter

import androidx.recyclerview.widget.RecyclerView
import com.example.to_do.Model.ToDoModel
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.example.to_do.R
import android.widget.CheckBox
import com.example.to_do.MainActivity

class ToDoAdapter(private val activity: MainActivity) : RecyclerView {
    private var todoList: List<ToDoModel>? = null
    fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemView = LayoutInflater.from(parent.context)
            .inflate(R.layout.task_layout, parent, false)
        return ViewHolder(itemView)
    }

    fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = todoList!![position]
        holder.task.text = item.task
        holder.task.isChecked = toBoolean(item.status)
    }

    fun getItemCount(): Int {
        return todoList!!.size
    }

    private fun toBoolean(n: Int): Boolean {
        return n != 0
    }

    fun setTasks(todolist: List<ToDoModel>?) {
        todoList = todolist
    }

    class ViewHolder internal constructor(view: View) : RecyclerView.ViewHolder(view) {
        var task: CheckBox

        init {
            task = view.findViewById(R.id.todoCheckBox)
        }
    }
}